### DISCUSSION TITLE
Building website on Swoole - domain best approach

### DISCUSSION THREAD

**Discussion by Draghmar** [swoole/swoole-src#4579] [General]
I was wondering lately how would perform typical website built on Swoole. Right now I'm mostly using it as a API server or for some background stuff, but still not related directly to the website and domains. 
So what would be the best approach to build website if you consider having to attach domain somehow. In Apache you just create vhost for that purpose. Is there way to do something similar? I mean, I guess it is possible to use some WWW server for handling domains and redirect traffic to backend. Like it's done with Apache and php-fpm for example. But wouldn't that hurt performance? And what about more than one domain?

**Comment by leocavalcante** [comment]
I think a good approach is to use a reverse proxy like Nginx on top of Swoole servers. Then you can easily configure domains and TLS termination.

**Reply by Draghmar** [reply]
Ok, but wouldn't that be a performance hit? I mean, this proxy would have to first manage the requests, right? I'm loosely thinking here.
If you're saying it's a good approach and not the best one ;), does it mean there are some other solutions?

**Reply by leocavalcante** [reply]
You will for sure pay a performance toll for that, BUT definitely it wouldn't be a bottleneck, you will not have slow requests because you are using a reverse proxy like Nginx before Swoole.

There are other approaches like trying to do the reverse proxying in the Swoole server it self, but it would require a lot of request handling based on domain etc, I don't think it worth the effort on 99.9% of the cases.
